universe/Universe.cpp

```diff
@@ -683,47 +683,60 @@ void Universe::InitMeterEstimatesAndDiscrepancies() {
     }
 }
 
-void Universe::UpdateMeterEstimates()
-{ UpdateMeterEstimates(INVALID_OBJECT_ID, false); }
-
-void Universe::UpdateMeterEstimates(int object_id, bool update_contained_objects) {
-    if (object_id == INVALID_OBJECT_ID) {
-        for (int obj_id : m_objects.FindExistingObjectIDs())
-            m_effect_accounting_map[obj_id].clear();
-        // update meters for all objects.  Value of updated_contained_objects is irrelivant and is ignored in this case.
-        UpdateMeterEstimatesImpl(std::vector<int>());// will cause it to process all existing objects
-        return;
-    }
-
-    // collect objects to update meter for.  this may be a single object, a group of related objects, or all objects
-    // in the (known) universe.  also clear effect accounting for meters that are to be updated.
-    std::set<int> objects_set;
-    std::list<int> objects_list;
-    objects_list.push_back(object_id);
+void Universe::UpdateMeterEstimates() {
+    for (int obj_id : m_objects.FindExistingObjectIDs())
+        m_effect_accounting_map[obj_id].clear();
+    // update meters for all objects.
+    UpdateMeterEstimatesImpl(std::vector<int>());
+}
+
+void Universe::UpdateMeterEstimates(int object_id, bool is_update_contained_objects) {
+    // ids of the object and all valid contained objects
+    std::unordered_set<int> all_ids;
+
+    // Collect objects to update meter for.  this may be a single object, a group of related
+    // objects, or all objects in the (known) universe.  also clear effect accounting for meters
+    // that are to be updated. Return true on success (all_ids is valid).
+    std::function<bool (int, int)> update_obj_and_contained_objs =
```

Comment:
Is this updating meters, clearing effect accounting, or collecting ids to update later? Naming seems inconsistent or incomplete.

Reply:
I refactored it so that it only collects ids and is called `collect_ids()`.